### PR TITLE
OpenAPI: add operationId

### PR DESCRIPTION
**Pull request by Fristi**
As defined here https://swagger.io/specification/

### CODE REVIEW

File: algebras/algebra/src/main/scala/endpoints/algebra/Endpoints.scala [hunk 1]
```diff
@@ -66,6 +66,7 @@ trait EndpointsWithCustomErrors extends Requests with Responses with Errors {
     * @param deprecated  Indicates whether this endpoint is deprecated or not
     */
   final class EndpointDocs private (
+      val operationId: Option[String],
```
[the remark this answers is not in the document]
Comment (Fristi) [reply algebras/algebra/src/main/scala/endpoints/algebra/Endpoints.scala:69]: Fixed

File: algebras/algebra/src/main/scala/endpoints/algebra/Endpoints.scala [hunk 2]
```diff
@@ -88,16 +89,34 @@ trait EndpointsWithCustomErrors extends Requests with Responses with Errors {
...
+      new EndpointDocs(
+        operationId,
+        summary,
+        description,
+        tags,
+        callbacks,
+        deprecated
+      )
+
+    def withOperationId(id: String): EndpointDocs =
```
[the remark this answers is not in the document]
Comment (Fristi) [reply algebras/algebra/src/main/scala/endpoints/algebra/Endpoints.scala]: Fixed

File: algebras/algebra/src/main/scala/endpoints/algebra/Endpoints.scala [hunk 3]
```diff
@@ -130,20 +149,22 @@ trait EndpointsWithCustomErrors extends Requests with Responses with Errors {
...
     def apply(): EndpointDocs =
-      new EndpointDocs(None, None, Nil, Map.empty, false)
+      new EndpointDocs(None, None, None, Nil, Map.empty, false)
 
     @deprecated(
       "Use `EndpointDocs().withXxx(...)` instead of `EndpointDocs(xxx = ...)`",
       "1.0.0"
     )
     def apply(
+        operationId: Option[String] = None,
```
[the remark this answers is not in the document]
Comment (Fristi) [reply algebras/algebra/src/main/scala/endpoints/algebra/Endpoints.scala]: Fixed, by removing it and setting the value to `None`

File: openapi/openapi/src/main/scala/endpoints/openapi/model/OpenApi.scala [hunk 4]
```diff
@@ -503,6 +504,7 @@ object Components {
 }
 
 final class Operation private (
+    val id: Option[String],
```
[the remark this answers is not in the document]
Comment (Fristi) [reply openapi/openapi/src/main/scala/endpoints/openapi/model/OpenApi.scala]: Good idea, fixed

File: openapi/openapi/src/main/scala/endpoints/openapi/model/OpenApi.scala [hunk 5]
```diff
@@ -561,6 +566,9 @@ final class Operation private (
       deprecated
     )
 
+  def withId(id: String): Operation =
```
[the remark this answers is not in the document]
Comment (Fristi) [reply openapi/openapi/src/main/scala/endpoints/openapi/model/OpenApi.scala]: Good idea, consistency is king. Fixed!

**Comment by Fristi** [comment]
Hopefully addressed your comments and added a test !

File: openapi/openapi/src/test/scala/endpoints/openapi/EndpointsTest.scala [hunk 6]
```diff
@@ -181,6 +181,22 @@ class EndpointsTest extends AnyWordSpec with Matchers with OptionValues {
     }
   }
 
+  "Operation documeation" should {
```
Comment (julienrf) [inline openapi/openapi/src/test/scala/endpoints/openapi/EndpointsTest.scala:184]:
```suggestion
  "Operation documentation" should {
```

**Comment by Fristi** [comment]
Is there a snapshot release available when we get this in? :)

**Comment by julienrf** [comment]
> Is there a snapshot release available when we get this in? :)

We (usually) don’t publish snapshots, but I plan to cut the next release within a week.

**Pull request merged**